### PR TITLE
Add doc on sonata_type_model sortable

### PR DESCRIPTION
**Pull request by qsomazzi**

### CODE REVIEW

File: Resources/doc/cookbook/recipe_sortable_sonata_type_model.rst [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+
+Part 1 : Update the data model configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The first thing to do is to update the Doctrine ORM configuration and create the join entity between ``User`` and ``Expectation``.
+We are going to call this join entity ``UserHasExpectations``.
+
+.. note::
+
+   We can't use a ``Many-To-Many`` relation here because the join entity will required an extra field to handle ordering.
```
Comment (OskarStark) [inline Resources/doc/cookbook/recipe_sortable_sonata_type_model.rst]: join -> joined ?

File: Resources/doc/cookbook/recipe_sortable_sonata_type_model.rst [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+Part 3 : Update admin classes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This is a very important part, the admin class **should** be created for the join entity. If you don't do that, the field will never display properly.
+So we are going to start by creating this ``UserBundle\Admin\UserHasExpectationsAdmin.php`` ...
+
+.. code-block:: php
+
+    <?php
+        namespace UserBundle\Admin;
```
Comment (OskarStark) [inline Resources/doc/cookbook/recipe_sortable_sonata_type_model.rst]: indention should be the same as ``<?php``

File: Resources/doc/cookbook/recipe_sortable_sonata_type_model.rst [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+Now update the ``UserBundle\Admin\UserAdmin.php`` by adding the ``sonata_type_model`` field.
+
+.. code-block:: php
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configureFormFields(FormMapper $formMapper)
+    {
+        ...
```
Comment (OskarStark) [inline Resources/doc/cookbook/recipe_sortable_sonata_type_model.rst]: ``// ...`` instead of ``...``

Comment (OskarStark) [inline Resources/doc/cookbook/recipe_sortable_sonata_type_model.rst]: or remove it

**Comment by qsomazzi** [comment]
@OskarStark corrected

**Comment by OskarStark** [comment]
Thank you :+1:

**Comment by OskarStark** [comment]
You have to add this here:
https://github.com/sonata-project/SonataAdminBundle/blob/master/Resources/doc/index.rst

**Comment by qsomazzi** [comment]
@OskarStark thx, I didn't seen it

**Comment by OskarStark** [comment]
Thank you @qsomazzi

**Comment by core23** [comment]
Build is failing https://travis-ci.org/sonata-project/SonataAdminBundle/builds/88751609

```
Warning, treated as error:
/home/travis/build/sonata-project/SonataAdminBundle/Resources/doc/reference/form_types.rst:101: WARNING: unknown document: recipe_sortable_sonata_type_model
```

**Comment by OskarStark** [comment]
Can you please have a look in the other files hin this can be fixed? Maybe add the cookbook folder as prefix @qsomazzi

**Comment by pulzarraider** [comment]
Doc link Typo is fixed in PR #3408

**Comment by qsomazzi** [comment]
Thanks @pulzarraider. I was too busy to check this last week :/

**Pull request merged**